src/evaluator/evaluator/evaluator.py

```diff
@@ -0,0 +1,223 @@
+
+        """!
+        Computes the perception's execution time
+        """
+
+        self.get_logger().info("Received perception")
+        self.end_time = datetime.datetime.now()
+        time_difference = self.end_time - self.start_time
+        execution_time = Float32()
+        execution_time.data = time_difference
+        self.perception_execution_time.publish(execution_time)
+
+
+    def compute_and_publish_perception(self):
+        """!
+        Computes perception metrics and publishes them.
+        """
+        mean_difference = Float32()
+        mean_difference.data = self.get_average_difference(self.perception_output, self.perception_ground_truth)
+
+        mean_squared_error = Float32()
+        mean_squared_error.data = self.get_mean_squared_error(self.perception_output, self.perception_ground_truth)
+
+        inter_cones_distance = Float32()
+        inter_cones_distance.data = self.get_inter_cones_distance(self.perception_output)
+
+        root_mean_squared_difference = Float32()
+        root_mean_squared_difference.data = sqrt(self.get_mean_squared_error(self.perception_output)
+)
+
+        # Publishes computed perception metrics
+        self.perception_mean_difference.publish(mean_difference)
+        self.perception_inter_cones_distance.publish(inter_cones_distance)
+        self.perception_mean_squared_difference.publish(mean_squared_error)
+        self.perception_root_mean_squared_difference.publish(root_mean_squared_difference)
+
+
+    @staticmethod
+    def get_average_difference(perception_output : list, perception_ground_truth : list):
+        """!
+        Computes the average difference between perception output and ground truth cones.
+        
+        Args:
+            perception_output (list): List of perceived cones.
+            perception_ground_truth (list): List of ground truth cones.
+
+        Returns:
+            float: Average difference between perception output and ground truth cones.
+        """
+        sum_error : float = 0
+        count : int = 0
+
+        if (len(perception_output) == 0):
+            return float('inf')
+        
+        if (len(perception_ground_truth) == 0):
+            raise ValueError("No ground truth cones provided for computing average difference.")
+
+        for perception_cone in perception_output:
+            min_distance : float = np.linalg.norm(perception_cone - perception_ground_truth[0])
+
+            for ground_truth_cone in perception_ground_truth:
+                distance : float = np.linalg.norm(perception_cone - ground_truth_cone)
+                if distance < min_distance:
+                    min_distance = distance
+
+            sum_error += min_distance
+            count += 1
+        
+        average = sum_error / count
+        return average
+    
+    @staticmethod
+    def get_mean_squared_error(perception_output: list, perception_ground_truth: list):
+        """!
```

Comment:
Maybe make this function more generic so that it can be used for other nodes' outputs, or if that's not possible maybe change the name so that it is specific to perception (this applies to get_average_difference as well), maybe just get_mean_squared_error_perception